src/kakarot/library.cairo

```diff
@@ -206,9 +206,54 @@ namespace Kakarot {
         // TODO: TEMPORARY SOLUTION FOR HACK-LISBON !!!
         let (_, low) = split_felt(contract_address);
         local mock_evm_address = 0xAbdE100700000000000000000000000000000000 + low;
+
         // Save address of new contracts
         let (reg_address) = registry_address.read();
         IRegistry.set_account_entry(reg_address, contract_address, mock_evm_address);
+
+        // Generate instructions set
+        let instructions: felt* = EVMInstructions.generate_instructions();
+
+        // Prepare execution context
+        let (empty_array: felt*) = alloc();
+        tempvar call_context: model.CallContext* = new model.CallContext(
+            bytecode=bytecode,
+            bytecode_len=bytecode_len,
+            calldata=empty_array,
+            calldata_len=0,
+            value=0,
+            );
+        let (local contract_bytecode: felt*) = alloc();
+        let stack: model.Stack* = Stack.init();
+        let memory: model.Memory* = Memory.init();
+        tempvar ctx: model.ExecutionContext* = new model.ExecutionContext(
+            call_context=call_context,
+            program_counter=0,
+            stopped=FALSE,
+            return_data=contract_bytecode,
+            return_data_len=0,
+            stack=stack,
+            memory=memory,
+            gas_used=0,
+            gas_limit=0,
+            intrinsic_gas_cost=0,
+            starknet_address=contract_address,
+            evm_address=mock_evm_address,
+            );
+
+        // Compute intrinsic gas cost and update gas used
+        let ctx = ExecutionContext.compute_intrinsic_gas_cost(ctx);
+
+        // Start execution
+        let ctx = run(instructions, ctx);
+
+        // Update contract bytecode with execution result
+        IEvmContract.write_bytecode(
+            contract_address=contract_address,
+            bytecode_len=ctx.return_data_len,
+            bytecode=ctx.return_data,
```

Comment:
I'm not sure I fully understand the flow at stake:
We give as input of this function (deploy) `bytecode_len: felt, bytecode: felt*` as params
This bytecode is the bytecode of the constructor of the smart contract?

This bytecode is literally the code of the constructor for that specific smart contract, and its return value is the smart contract's code without the constructor inside it?

Reply:
so basically a contract has its own bytecode.
However, when you want to deploy a contract, you send a transaction with a differente bytecode because you need to run the constructor of this contract.

Eventually the deploy transaction is special as it creates a new bytecode that somehow concat the contract bytecode and with an initial bytecode standing for the call to the constructor. This new bytecode is send as calldata to address `0x0`.

For this special transaction, the EVM runs the calldata as is, ie process the calldata as a bytecode. The output of this initial run is the bytecode of the initialized contract. This output is eventually stored as the final contract bytecode.

You will see it clearer in the tests with the 2 PRs merged and the use of `web3py`.

Reply:
Awesome! Thanks